subsys/bluetooth/controller/ll_sw/ull_llcp.c

```diff
@@ -1019,6 +1034,114 @@ uint8_t ull_cp_conn_update(struct ll_conn *conn, uint16_t interval_min, uint16_t
 	return BT_HCI_ERR_SUCCESS;
 }
 
+#if defined(CONFIG_BT_CTLR_SYNC_TRANSFER_SENDER)
+uint8_t ull_cp_periodic_sync(struct ll_conn *conn, struct ll_sync_set *sync,
+			     struct ll_adv_sync_set *adv_sync, uint16_t service_data)
+{
+	struct pdu_adv_sync_info *si;
+	struct proc_ctx *ctx;
+	uint8_t *access_addr;
+	const uint8_t *adva;
+	uint16_t interval;
+	uint8_t *chan_map;
+	uint8_t *crc_init;
+	uint8_t addr_type;
+	uint8_t si_sca;
+	uint8_t sid;
+	uint8_t phy;
+
+	/* Exactly one of the sync and adv_sync pointers should be non-null */
+	LL_ASSERT((!adv_sync && sync) || (adv_sync && !sync));
+
+	if (!feature_peer_periodic_sync_recv(conn)) {
+		return BT_HCI_ERR_UNSUPP_REMOTE_FEATURE;
+	}
+
+	ctx = llcp_create_local_procedure(PROC_PERIODIC_SYNC);
+	if (!ctx) {
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
+	if (sync) {
+		chan_map = sync->lll.chm[sync->lll.chm_first].data_chan_map;
+		si_sca = sync->lll.sca;
+		access_addr = sync->lll.access_addr;
+		crc_init = sync->lll.crc_init;
+		sid = sync->sid;
+		phy = sync->lll.phy;
+		interval = sync->interval;
+
+		addr_type = sync->peer_id_addr_type;
+		if (sync->peer_addr_resolved) {
+			uint8_t rl_idx;
+
+			/* peer_id_addr contains the identity address; Get the peers RPA */
+
+			rl_idx = ull_filter_rl_find(addr_type, sync->peer_id_addr, NULL);
+
+			/* A resolved address must be present in the resolve list */
+			LL_ASSERT(rl_idx < ll_rl_size_get());
+
+			/* Generate RPAs if required */
+			ull_filter_rpa_update(false);
+
+			/* Note: Since we need the peers RPA, use tgta_get */
+			adva = ull_filter_tgta_get(rl_idx);
+		} else {
+			adva = sync->peer_id_addr;
+		}
+	} else {
+		struct ll_adv_set *adv;
+		struct pdu_adv *adv_pdu;
+
+		chan_map = adv_sync->lll.chm[adv_sync->lll.chm_first].data_chan_map;
+		si_sca = lll_clock_sca_local_get();
+		access_addr = adv_sync->lll.access_addr;
+		crc_init = adv_sync->lll.crc_init;
+		phy = adv_sync->lll.adv->phy_s;
+		interval = adv_sync->interval;
+
+		adv = HDR_LLL2ULL(adv_sync->lll.adv);
+		sid = adv->sid;
+
+		/* Pull AdvA from pdu */
+		adv_pdu = lll_adv_sync_data_curr_get(&adv_sync->lll);
+		addr_type = adv_pdu->tx_addr;
+		/* Note: AdvA is mandatory for AUX_SYNC_IND and at the start of the ext. header */
+		adva = adv_pdu->adv_ext_ind.ext_hdr.data;
+	}
+
+	/* Store parameters in corresponding procedure context */
+	ctx->data.periodic_sync.sync_handle = sync ? ull_sync_handle_get(sync) :
+							INVALID_SYNC_HANDLE;
+	ctx->data.periodic_sync.adv_handle = adv_sync ? ull_adv_sync_handle_get(adv_sync) :
+							INVALID_ADV_HANDLE;
+	ctx->data.periodic_sync.id = service_data;
```

Comment:
Would it make sense to use `service_data` instead of `id` in the `periodic_sync` struct?

Reply:
The only reason it is called ID is because of the naming in the Spec:

"ID shall be set to an identifier provided by the Host. This is for use by higher-level protocols and its value is not specified or used by this specification."

Reply:
Aha, interesting. I'm having trouble finding where it's defined that the `id` here is the same value as the `service_data` provided by the host. I believe it's correct that it is, but haven't found where that's explicitly stated

Reply:
I haven't seen it explicitly stated anywhere in the Spec. However, it is the only host parameter in `HCI_LE_­Periodic_­Advertising_­Sync_­Transfer` and `HCI_LE_­Periodic_­Advertising_­Sync_­Transfer` with the correct range and description, but it is not explicitly stated anywhere.